lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -2303,7 +2303,7 @@ using namespace Js;
                     }
                     if (setterValueOrProxy)
                     {
-                        if (!WithScopeObject::Is(receiver) && info->GetPropertyRecordUsageCache())
+                        if (!WithScopeObject::Is(receiver) && info->GetPropertyRecordUsageCache() && !JavascriptOperators::IsUndefinedAccessor(setterValueOrProxy, requestContext))
```

Comment:
Is it possible to have a default setter added to an object such that it uses the undefined accessor of a different context? In node I'm thinking something like

```js
var o = {};
var ctx = {o: o}
vm.createContext(ctx);
vm.runInContext("Object.defineProperty(o, 'x', {get: () => 1, configurable: true, enumerable: true})", ctx);
o.x = 2;
```

In that script, the `Object.defineProperty` line is invoked in a different script context using Object from a different library. Will this change detect that the write to `o.x` back in the original context is using the defaulted accessor?

Reply:
That shouldn't be an issue, because we don't support caching cross-context (`CacheOperators::CanCachePropertyWrite`).